teachertool/src/components/CatalogOverlay.tsx

```diff
@@ -78,19 +78,27 @@ const CatalogList: React.FC = () => {
     const { state: teacherTool } = useContext(AppStateContext);
 
     const recentlyAddedWindowMs = 500;
-    const [recentlyAddedIds, setRecentlyAddedIds] = useState<string[]>([]);
+    const [recentlyAddedIds, setRecentlyAddedIds] = useState<pxsim.Map<boolean>>({}); // Id -> bool, True if recently added. False (or not in list) otherwise.
 
     const criteria = useMemo<CatalogCriteria[]>(
         () => getCatalogCriteria(teacherTool),
         [teacherTool.catalog, teacherTool.rubric]
     );
 
+    function updateRecentlyAddedValue(id: string, value: boolean) {
+        setRecentlyAddedIds(prevState => {
+            const newState = { ...prevState };
```

Comment:
Do you need to make this newState variable? If the updater function is going to modify the previous state, adding/updating the `id` entry of that state should work, right?

Reply:
Oh, right, makes sense!